src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -721,14 +723,14 @@ export class ResultsViewPageStore {
             this.samples
         ],
         invoke: () => {
-            if (this.genes.result) {
-                return Promise.resolve(_.reduce(_.uniq(this.genes.result.map(g => g.chromosome)), (map: { [chromosome: string]: MobxPromise<CopyNumberSeg[]> }, chromosome: string) => {
+            if (this.referenceGenes.result) {
+                const uniqueReferenceGenes = _.uniq(this.referenceGenes.result.map(g => g.chromosome));
+                return Promise.resolve(uniqueReferenceGenes.reduce(
```

Comment:
lets assign result of reduce to local variable first and then resolve it.  hard to read